src/v/cluster/data_migration_backend.cc

```diff
@@ -428,26 +470,196 @@ backend::do_topic_work(model::topic_namespace nt, topic_work tw) noexcept {
+        co_return errc::success;
+    }
+    if (unmount_res != errc::success) {
+        vlog(dm_log.warn, "failed to unmount topic {}: {}", nt, unmount_res);
+        co_return unmount_res;
+    }
+    co_return co_await retry_loop(rcn, [this, &nt, &rcn] {
+        return _topics_frontend.delete_topic_after_migration(
+          nt, rcn.get_deadline());
+    });
+}
+
+ss::future<errc> backend::create_topic(
+  const model::topic_namespace& local_nt,
+  const std::optional<model::topic_namespace>& original_nt,
+  const std::optional<cloud_storage_location>& storage_location,
+  retry_chain_node& rcn) {
+    // download manifest
+    const auto& bucket_prop = cloud_storage::configuration::get_bucket_config();
+    auto maybe_bucket = bucket_prop.value();
+    if (!maybe_bucket) {
+        co_return errc::topic_operation_error;
+    }
+    cloud_storage::topic_manifest_downloader tmd(
+      cloud_storage_clients::bucket_name{*maybe_bucket},
+      storage_location ? std::make_optional(storage_location->hint)
+                       : std::nullopt,
+      original_nt.value_or(local_nt),
+      _cloud_storage_api->get()); // checked in frontend::data_migrations_active
+    // todo: is it correct to rely on it checked in frontend?
+    // todo: configure timeout and backoff
+    auto backoff = std::chrono::duration_cast<model::timestamp_clock::duration>(
+      rcn.get_backoff());
+    cloud_storage::topic_manifest tm;
+    auto download_res = co_await tmd.download_manifest(
+      rcn, rcn.get_deadline(), backoff, &tm);
+    if (
+      !download_res.has_value()
+      || download_res.assume_value()
+           != cloud_storage::find_topic_manifest_outcome::success) {
+        vlog(
+          dm_log.warn,
+          "failed to download manifest for topic {}: {}",
+          original_nt.value_or(local_nt),
+          download_res);
+        co_return errc::topic_operation_error;
+    }
+    auto maybe_cfg = tm.get_topic_config();
+    if (!maybe_cfg) {
+        co_return errc::topic_invalid_config;
+    }
+    // todo: unify with recovery?
+    cluster::topic_configuration topic_to_create_cfg(
+      local_nt.ns,
+      local_nt.tp,
+      maybe_cfg->partition_count,
+      maybe_cfg->replication_factor);
+    auto& topic_properties = topic_to_create_cfg.properties;
+    auto manifest_props = maybe_cfg->properties;
+
+    topic_properties.compression = manifest_props.compression;
+    topic_properties.cleanup_policy_bitflags
+      = manifest_props.cleanup_policy_bitflags;
+    topic_properties.compaction_strategy = manifest_props.compaction_strategy;
+
+    topic_properties.retention_bytes = manifest_props.retention_bytes;
+    topic_properties.retention_duration = manifest_props.retention_duration;
+
+    topic_properties.retention_local_target_bytes = tristate<size_t>{
+      config::shard_local_cfg()
+        .cloud_storage_recovery_temporary_retention_bytes_default};
```

Comment:
i am wondering if this is correct, do we eventually set the correct value for this property ?